src/components/NavigationBar/index.tsx

```diff
@@ -62,26 +71,124 @@ export const navLinks = (
   }
 ];
 
-const NavigationBar = ({
-  mobile,
-  signout,
-  toggle,
-  userName
-}: NavigationProps) => {
+const systemLinks = (
+  flags: Flags,
+  userGroups: string[],
+  isUserSet: boolean
+) => [
+  {
+    link: '/systems?table-type=all-systems',
+    label: 'allCMSSystems',
+    isEnabled: true
+  },
+  {
+    link: '/systems?table-type=my-systems',
+    label: 'mySystems',
+    isEnabled: true
+  },
+  {
+    link: '/systems?table-type=bookmarked-systems',
+    label: 'bookmarkedSystems',
+    isEnabled: true
+  },
+  {
+    link: '/system/request-type',
+    label: 'addNewSystem',
+    isEnabled: true
+  }
+];
+
+// Handler for updating state for any NavDropDownButton
+const onToggle = (
+  index: number,
+  setIsOpen: React.Dispatch<React.SetStateAction<boolean[]>>
+): void => {
+  setIsOpen(prevIsOpen => {
+    const newIsOpen = [false];
+    newIsOpen[index] = !prevIsOpen[index];
+    return newIsOpen;
+  });
+};
+
+const NavigationBar = ({ signout, userName }: NavigationProps) => {
   const { t } = useTranslation();
+
   const flags = useFlags();
+
   const { groups, isUserSet } = useSelector((state: AppState) => state.auth);
 
+  const mobileNavRef = useRef<HTMLDivElement | null>(null);
+  const systemNavRef = useRef<HTMLDivElement | null>(null);
+
+  const isMobile = useCheckResponsiveScreen('tablet');
+
+  const location = useLocation();
+
+  // Controls mobile side nav
+  const [expanded, setExpanded] = useState(false);
+
+  // Controls NavDropDownButton state for (currently one) any nav dropdown items
+  const [isOpen, setIsOpen] = useState([false]);
```

Comment:
why an array? it looks like in the future if we have multiple nav dropdowns, this can look like `useState([false, false, true, ...])`
could we make this an object instead?

`useState({nav1: false, nav2: false, nav3: true})` 

then we could avoid having to index when setting state?

Reply:
All good!